diff --git a/dist/client/components/router-reducer/fetch-server-response.js b/dist/client/components/router-reducer/fetch-server-response.js
index 0484557dae9de8d9b27049d56a3fb465d04ee373..cc1ac1abcf595cd036a91b582bf670b042f1819a 100644
--- a/dist/client/components/router-reducer/fetch-server-response.js
+++ b/dist/client/components/router-reducer/fetch-server-response.js
@@ -163,7 +163,7 @@ async function fetchServerResponse(url, options) {
         // If fetch fails handle it like a mpa navigation
         // TODO-APP: Add a test for the case where a CORS request fails, e.g. external url redirect coming from the response.
         // See https://github.com/vercel/next.js/issues/43605#issuecomment-1451617521 for a reproduction.
-        return originalUrl.toString();
+        return doMpaNavigation(originalUrl.toString());
     }
 }
 async function createFetch(url, headers, fetchPriority, shouldImmediatelyDecode, signal) {
